git-cliff/src/lib.rs

```diff
@@ -221,22 +236,13 @@ fn process_repository<'a>(
 	}
 
 	// Set the previous release if needed.
-	if args.latest || args.unreleased {
-		let sub = if args.latest || args.tag.is_some() {
-			2
-		} else {
-			1
+	if let Some((commit_id, version)) = first_tag {
+		let previous_release = Release {
+			commit_id: Some(commit_id.to_string()),
+			version: Some(version.to_string()),
+			..Release::default()
 		};
-		if let Some((commit_id, version)) =
-			tags.len().checked_sub(sub).and_then(|v| tags.get_index(v))
-		{
-			let previous_release = Release {
-				commit_id: Some(commit_id.to_string()),
-				version: Some(version.to_string()),
-				..Release::default()
-			};
-			releases[0].previous = Some(Box::new(previous_release));
-		}
+		releases[0].previous = Some(Box::new(previous_release));
```

Comment:
Why are we not checking if `args.latest` or `args.unreleased` is used? Does this mean this will override the previous release even though it is not needed?

Reply:
Maybe what I can do as an improvement, and that is probably what you mean, is to only set previous if not already set?

Reply:
Yeah, we can do that but I guess this is also fine.

Reply:
Updated the code with that check also, to avoid the unnecessary performance hit 👍